fixham/build/main.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"github.com/goyek/goyek/v2"
+
+	"github.com/projectcalico/fixham/pkg/bootstrap"
+)
+
+const packageName = "github.com/projectcalico/fixham"
+
+var _ = bootstrap.DefineCleanTask([]string{"./bin/*"}, nil, nil)
```

Comment:
I think we should think a bit more about how to structure this / build an API to use in other repositories - there are some global variables being initialized / written at different points in time that make this code tricky to follow, and prone to issues like nil pointer exceptions if things are executed in the wrong order. Getting the code structure as close to "right" as we can will have a huge positive impact on future development.

Here's some pretty rough pseudocode that hopefully illustrates one idea for how to organize this code. If it's not clear, let's hop on a Zoom call to discuss:

```go
package api

// Package represents a package within the projectcalico/calico repository that can have one or more
// tasks associated with it.
type Package interface {
  Path() string
  Tasks() []goyek.Task
}
```

```go
package main

func main() {
  // Load configuration from environment.
  cfg := LoadConfig() 

  // Initialize the package we want to execute tasks for.
  fixham := NewFixham(cfg)
  tasks.Register(fixham)
}
```

```go
package main

func NewFixham(cfg *config.Config) api.Package {
  return &fixham{...}
}

func (f *fixham) Clean() goyek.Task {
  return common.Clean(...)
}

func (f *fixham) Tasks() []goyek.Task {
  return []goyek.Task{f.Clean()}
}
```

```go
package tasks

func Register(p api.Package) error {
  for _, task := range p.Tasks() {
    goyek.Define(task)
  }
}
```